src/components/timeline/Selection.js

```diff
@@ -431,7 +433,13 @@ class TimelineRulerAndSelection extends React.PureComponent<Props> {
                 : undefined,
             left: hoverLocation === null ? '0' : `${hoverLocation}px`,
           }}
-        />
+        >
+          <span className="timelineSelectionOverlayTime">
+            {mouseTimePosition !== null
+              ? getFormattedTimeLength(mouseTimePosition - zeroAt)
```

Comment:
I believe it should use a similar logic for the precision than the ruler, because otherwise it's not useful when zoomed in a lot (except at the start of the profile).
The logic is in timeline/Ruler.js, but here is what is I believe a good enough approximation:
```js
const range = committedRange.end - committedRange.start;
const exponent = Math.floor(Math.log10(range));
const digits = Math.max(0, -(exponent - 4))
const timeString = (mouseTimePosition - zeroAt).toFixed(digits);
```
I haven't tried it though, tell me how that looks.

Reply:
I actually care more about showing something useful for long profiles (ie. not hundreads or thousands of seconds, but something in minutes and seconds) than about the high precision when zoomed into extremely short intervals.
Of course having both would be best, but that might require a lot more, like an alternative version of `formatNumber`(or adding an extra parameter to it) to specify the required smallest unit to display.

The numbers displayed in the ruler are currently difficult to read for long profiles. That's something I would like to improve someday. Maybe I could fix the numbers displayed in the ruler and polish the number in this new overlay at the same time?

A short term alternative if you really care about short intervals could be to switch to the implementation you suggested when the commited range lasts less than a few seconds. Not sure if that would be better or more confusing.

Reply:
I believe you could do:
```js
const range = committedRange.end - committedRange.start;
const exponent = Math.floor(Math.log10(range));
const digits = Math.max(0, -(exponent - 4))
const timeString =
  digits > 0
  ? ((mouseTimePosition - zeroAt)/1000).toFixed(digits) + "s"
  : getFormattedTimeLength(mouseTimePosition - zeroAt);
}
```